File: 백준 14938 FW 서강그라운드.js
const fs = require('fs');
let [n, m, r, ...input] = fs.readFileSync('/dev/stdin').toString().trim().split(/\s+/).map(Number);
const items = input.splice(0, n);
const INF = Number.MAX_SAFE_INTEGER;
const MAX = 101;
const dist = Array.from({ length: MAX }, () => Array(MAX).fill(INF));

for (let i = 1; i <= n; i++) {
    dist[i][i] = 0;
}

for (let i = 0; i < r; i++) {
    const [a, b, l] = [input[i*3], input[i*3+1], input[i*3+2]];
    dist[a][b] = l;
    dist[b][a] = l;
}

function floydWarshall(n, dist) {
    for (let k = 1; k <= n; ++k) {
        for (let i = 1; i <= n; ++i) {
            for (let j = 1; j <= n; ++j) {
                if (dist[i][k] + dist[k][j] < dist[i][j]) {
                    dist[i][j] = dist[i][k] + dist[k][j];
                }
            }
        }
    }
}

floydWarshall(n, dist);

let maxItems = 0;
for (let i = 1; i <= n; ++i) {
    let total = 0;
    for (let j = 1; j <= n; ++j) {
        if (dist[i][j] <= m) {
            total += items[j-1]; // Adjusting index for items array
        }
    }
    maxItems = Math.max(maxItems, total);
}

console.log(maxItems);

File: 백준 14938 FW 서강그라운드.py
import sys

input = sys.stdin.readline


def floydWarshall(n, dist):
    for k in range(1, n + 1):
        for i in range(1, n + 1):
            for j in range(1, n + 1):
                dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])


n, m, r = map(int, input().split())
items = [0] + list(map(int, input().split()))

INF = float("inf")
dist = [[INF] * (n + 1) for _ in range(n + 1)]

for i in range(1, n + 1):
    dist[i][i] = 0

for _ in range(r):
    a, b, l = map(int, input().split())
    dist[a][b] = l
    dist[b][a] = l

floydWarshall(n, dist)

max_items = 0
for i in range(1, n + 1):
    total = 0
    for j in range(1, n + 1):
        if dist[i][j] <= m:
            total += items[j]
    max_items = max(max_items, total)

print(max_items)
